src/utils/IntlWrapper.ts

```diff
@@ -319,28 +220,27 @@ export class IntlDateTime {
     return this.locale;
   }
 
-  getStyle() {
-    return this.dateTimeFormatStyle;
-  }
-
-  getRelative(
-    input: number | durationFormat,
-    style: durationStyle = 'Duration',
-    convertFrom?: keyof typeof dateUnitToMs,
-    format?: durationFormatStyle,
-  ) {
-    const duration = new IntlDuration(input, style, convertFrom, format, this.locale);
-    return duration.getText();
+  getDateTimeText(style: Intl.DateTimeFormatOptions = { dateStyle: 'medium' }) {
+    if (!isValidDate(this.date)) return '';
+    return new Intl.DateTimeFormat(this.locale, style).format(this.date);
   }
 
-  getText() {
-    this.format();
-    return this.text;
+  getRelativeNowText(style: durationStyle = 'Duration', format?: durationFormatStyle) {
+    if (!this.isValidDate()) return '';
+    const relative = new IntlDuration().setLocale(this.locale);
+    relative.setTimestamp(this.date.getTime(), style);
+    const duration = relative.getDuration();
+    if (!duration) return '';
+    if (checkForNow(duration)) return api.storage.lang('bookmarksItem_now');
+    return relative.getRelativeText(format);
   }
 
-  getRange(from: Date | number | string, to: Date | number | string) {
-    const range = new IntlRange(from, to, this.dateTimeFormatStyle, this.locale);
-    return range.getText();
+  getRelativeNowFriendlyText(style: durationStyle = 'Duration', format?: durationFormatStyle) {
+    const relative = new IntlDuration().setLocale(this.locale);
+    relative.setTimestamp(this.date.getTime(), style);
+    const duration = relative.getDuration();
+    if (!duration) return '';
+    return relative.getRelativeText(format);
   }
```

Comment:
Not what I was expecting of the function. Was thinking of something more in this direction. Specially as yours has a lot of duplicate code and does not return what I was expecting

```ts
    getRelativeNowText(style: durationStyle = 'Duration', format?: durationFormatStyle) {
      if (!this.isValidDate()) return '';
      const relative = new IntlDuration(this.locale);
      relative.setTimestamp(this.date.getTime(), style);
      return relative.getRelativeText(format);
    }
    getRelativeTimeText(style: durationStyle = 'Duration', format?: durationFormatStyle) {
        if (this.isNow()) return api.storage.lang('bookmarksItem_now');
        timeString = this.getRelativeNowText(style, format);
        return this.isFuture() ? timeString :  api.storage.lang('bookmarksItem_agoxyz',[timeString ]);
    }
```

Reply:
Oh I see, thanks for pinpointing this
Btw we already have `prediction_Episode_****` and `prediction_Last_****` in `progress.ts` which returns strings with "ago", do we even need them? I guess I just delete those lines in `progress.ts` to avoid double "ago" and remove `getLastText` and `getPredictionText` since we already returning final text from `IntlDatetime`

Reply:
As it was not happening in progress at all. I though you wanted to first do the structure and I did not point it out. But probably because of our bad progress structure. getLast is what is used nearly anywhere and can directly use FriendlyText. getLastText is used very seldom, was thinking of removing it, that is why it is ok to just implement it again there, as we do not really care. 

I would not handle the `getLastText` inside of the getRelativeNowFriendlyText, as it is not that general and we do not really care.

```ts
// progress.ts
  getLast(): string {
    const timestamp = Number(this.getLastTimestamp());
    return new IntlDateTime(timestamp).getRelativeNowFriendlyText('Progress'); // returns 2d ago
  }
  getLastText(): string {
    const dt = new IntlDateTime(Number(timestamp));
    if (dt.isNow()) return api.storage.lang('bookmarksItem_now');
    return dt.isFuture() ? timeString :  api.storage.lang(`prediction_Last_${this.type}`, [timeString ]); // returns Last episode was 2d ago
  }
```

Reply:
Done